javascript/javascript2/week3/index.js

```diff
@@ -0,0 +1,193 @@
+
+//Check if we have double clicked on the page. 
+//A double click is defined by two clicks within 0.5 seconds. 
+//If a double click has been detected, log out the text: "double click!"
+var eventFire = false;
+const detector = document.querySelector('aside').addEventListener("dblclick", function (){
+    eventFire = true ;
+    console.log("fire")
+    if(eventFire) {
+        alert('Event Fire')
+        document.getElementById("detector-text").innerHTML = "double click detected";
+    } else {
+        console.log("erreur")
+    }
+})
+
+//Create a function called jokeCreator that has three parameters: shouldTellFunnyJoke - boolean, 
+//logFunnyJoke - function 
+//and logBadJoke - function.
+// If you set shouldTellFunnyJoke to true it should call
+// the logFunnyJoke function that should log out a funny joke. And vice versa.
+const badJoke = () => {
+    console.log("Who’s there?Interrupting cow Interrupting c– MOO!")
+} 
+const funnyJoke = () => {
+    console.log("What do you call a boomerang that won’t come back it is  a stick hahahah")
+} 
+
+const jokeCreator = ( shouldTellFunnyJoke, logFunnyJoke, logBadJoke) => {
+    if(shouldTellFunnyJoke){
+        logFunnyJoke()
+    }else{
+        logBadJoke()
+    }
+}
+jokeCreator(true, funnyJoke, badJoke)
+
+//Function as a variable    
+const first_function = (text) => {
+    console.log(text)
+};
+
+const second_function = (text) => {
+    console.log(text)
+};
+
+const third_function = (text) => {
+    console.log(text)
+};
+
+var arrayOfFunctions = [
+    first_function,
+    second_function,
+    third_function
+]
+
+for (let i = 0; i < arrayOfFunctions.length; i++) {
+    arrayOfFunctions[i](`the ${i} call function`);
+}
+
+//Create a function as a const and try creating a function normally. 
+//Call both functions.
+const doSomething = () => {
+    console.log("function as a const")
+}
+function normalFunction () {
+    console.log("normal function")
+}
+doSomething();
+normalFunction();
+
+//Create an object that has a key whose value is a function. 
+//Try calling this function.
+const country = {
+    location: function() {
+      console.log("my country is Tunisia")
+    }
+};
+country.location()
+
+//extra 
+const person = {
+    firstName: "John",
+    lastName: "Doe",
+    id: 5566,
+    fullName: function() {
+      return this.firstName + " " + this.lastName;
+    }
+};
+
+//The fastest presser in this realm
+var countS = 0;
+var countL = 0;
+
+document
+    .getElementById("button-game")
+    .addEventListener("click", function(){
+        let time = document.getElementById("seconds").value;
+    setTimeout(() => {const end = document.getElementById("end");
+    end.innerHTML = "Game Over";}, time * 1000);
+})
+
+
+document.getElementById("sPresser").addEventListener("keydown", function(event){
+    if (event.key == "s") {
+        countS++;
+        const numberOfPress = document.getElementById("startS");
+        numberOfPress.innerHTML = countS;
+    }
+})
+
```

Comment:
Maybe you can move the fast presser to another file, which will makes it much more easier to go through